src/github.com/getlantern/flashlight/util/http.go

```diff
@@ -35,6 +36,18 @@ func HTTPClient(rootCA string, proxyAddr string) (*http.Client, error) {
 // validate the server's certificate on TLS connections against that RootCA. If
 // proxyAddr is specified, the client will proxy through the given http proxy.
 func httpClient(rootCA string, proxyAddr string, persistent bool) (*http.Client, error) {
+
+	log.Debugf("Waiting for proxy server...")
+
+	// Waiting for proxy server to came online.
+	err := waitforserver.WaitForServer("tcp", proxyAddr, 60*time.Second)
+	if err != nil {
+		// Instead of finishing here we just log the error and continue, the client
+		// we are going to create will surely fail when used and return errors,
+		// those errors should be handled by the code that depends on such client.
+		log.Errorf("Proxy never came online at %v: %q", proxyAddr, err)
+	}
+
```

Comment:
Yeah I like the general thinking here @xiam . Two issue I see though:

1. This method is at least designed to not require a proxy, so I would move this after line 78 ```if proxyAddr != "" {```
1. If the proxy is not a localhost address, this might not be what we want (it would probably make more sense to just let it fail). We could use the IP IsLoopback method to check that before making this call -- http://golang.org/pkg/net/#IP.IsLoopback